src/ParallelAlgorithms/LinearSolver/Schwarz/ComputeTags.hpp

```diff
@@ -0,0 +1,177 @@
+#include "ParallelAlgorithms/LinearSolver/Schwarz/Weighting.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace LinearSolver::Schwarz::Tags {
+
+/// The number of points a neighbor's subdomain extends into the element
+template <size_t Dim, typename OptionsGroup>
+struct IntrudingExtentsCompute : db::ComputeTag,
+                                 IntrudingExtents<Dim, OptionsGroup> {
+  using base = IntrudingExtents<Dim, OptionsGroup>;
+  using return_type = typename base::type;
+  using argument_tags =
+      tmpl::list<domain::Tags::Mesh<Dim>, MaxOverlap<OptionsGroup>>;
+  static constexpr void function(
+      const gsl::not_null<return_type*> intruding_extents,
+      const Mesh<Dim>& mesh, const size_t max_overlap) noexcept {
+    for (size_t d = 0; d < Dim; ++d) {
+      gsl::at(*intruding_extents, d) =
+          LinearSolver::Schwarz::overlap_extent(mesh.extents(d), max_overlap);
+    }
+  }
+};
+
+/// The width in element-logical coordinates that a neighbor's subdomain extends
+/// into the element
+template <size_t Dim, typename OptionsGroup>
+struct IntrudingOverlapWidthsCompute
+    : db::ComputeTag,
+      IntrudingOverlapWidths<Dim, OptionsGroup> {
+  using base = IntrudingOverlapWidths<Dim, OptionsGroup>;
+  using return_type = typename base::type;
+  using argument_tags =
+      tmpl::list<domain::Tags::Mesh<Dim>, IntrudingExtents<Dim, OptionsGroup>>;
+  static constexpr void function(
+      const gsl::not_null<return_type*> intruding_overlap_widths,
+      const Mesh<Dim>& mesh,
+      const std::array<size_t, Dim>& intruding_extents) noexcept {
+    for (size_t d = 0; d < Dim; ++d) {
+      const auto& collocation_points =
+          Spectral::collocation_points(mesh.slice_through(d));
+      gsl::at(*intruding_overlap_widths, d) =
+          LinearSolver::Schwarz::overlap_width(gsl::at(intruding_extents, d),
+                                               collocation_points);
+    }
+  }
+};
+
+/// Weighting field for data on the element
+template <size_t Dim, typename OptionsGroup>
+struct ElementWeightCompute : db::ComputeTag, Weight<OptionsGroup> {
+  using base = Weight<OptionsGroup>;
+  using return_type = tmpl::type_from<base>;
+  using argument_tags =
+      tmpl::list<domain::Tags::Element<Dim>,
+                 domain::Tags::Coordinates<Dim, Frame::Logical>,
+                 IntrudingOverlapWidths<Dim, OptionsGroup>,
+                 MaxOverlap<OptionsGroup>>;
+  static constexpr void function(
+      const gsl::not_null<return_type*> element_weight,
+      const Element<Dim>& element,
+      const tnsr::I<DataVector, Dim, Frame::Logical>& logical_coords,
+      const std::array<double, Dim>& intruding_overlap_widths,
+      const size_t max_overlap) noexcept {
+    // For max_overlap > 0 all overlaps will have non-zero extents, so we don't
+    // need to check their extents are non-zero individually
+    if (LIKELY(max_overlap > 0)) {
```

Comment:
- Yes, that's always the case on an LGL mesh, because it has at least 2 points per dimension. I clarified the comment.
- The `element_weight` function only depends on the overlap _widths_ currently, so I would prefer not to add a dependence on the num overlap points just for this short-circuiting. I think it's OK to have the compute item handle that. Basically the interface is "don't try to do weighting at all if the overlap is empty".